`test_rescaleData` fails on aarch64

### Short description
`test_rescaleData` fails on aarch64:
```
[  551s] =================================== FAILURES ===================================
[  551s] _______________________________ test_rescaleData _______________________________
[  551s] [gw2] linux -- Python 3.8.16 /usr/bin/python3.8
[  551s] 
[  551s]     def test_rescaleData():
[  551s]         dtypes = map(np.dtype, ('ubyte', 'uint16', 'byte', 'int16', 'int', 'float'))
[  551s]         for dtype1 in dtypes:
[  551s]             for dtype2 in dtypes:
[  551s] >               data = (np.random.random(size=10) * 2**32 - 2**31).astype(dtype1)
[  551s] E               RuntimeWarning: invalid value encountered in cast
```

### Code to reproduce
Run tests on openSUSE Tumbleweed aarch64.

### Expected behavior
Test should pass.

### Real behavior
Test fails.

### Tested environment(s)

 * PyQtGraph version: 0.12.4 and 0.13.1
 * Qt Python binding: 6.4.2
 * Python version: 3.8.16
 * NumPy version: 1.24.1
 * Operating system: openSUSE Tumbleweed aarch64
 * Installation method: from source

--- a/tests/test_functions.py
+++ b/tests/test_functions.py
@@ -118,10 +118,15 @@ def test_subArray():
     
     
 def test_rescaleData():
+    rng = np.random.default_rng(12345)
     dtypes = map(np.dtype, ('ubyte', 'uint16', 'byte', 'int16', 'int', 'float'))
     for dtype1 in dtypes:
         for dtype2 in dtypes:
-            data = (np.random.random(size=10) * 2**32 - 2**31).astype(dtype1)
+            if dtype1.kind in 'iu':
+                lim = np.iinfo(dtype1)
+                data = rng.integers(lim.min, lim.max, size=10, dtype=dtype1, endpoint=True)
+            else:
+                data = (rng.random(size=10) * 2**32 - 2**31).astype(dtype1)
             for scale, offset in [(10, 0), (10., 0.), (1, -50), (0.2, 0.5), (0.001, 0)]:
                 if dtype2.kind in 'iu':
                     lim = np.iinfo(dtype2)

